src/core/Akka.Remote/Endpoint.cs

```diff
@@ -609,9 +635,10 @@ private void HandleSend(EndpointManager.Send send)
             {
                 var sequencedSend = send.Copy(NextSeq());
                 TryBuffer(sequencedSend);
-                //If we have not confirmed the remote UID we cannot transfer the system message at this point, so just buffer it.
-                // GotUid will kick ResendAll causing the messages to be properly written.
-                if (UidConfirmed) _writer.Tell(sequencedSend);
+                // If we have not confirmed the remote UID we cannot transfer the system message at this point just buffer it.
+                // GotUid will kick ResendAll() causing the messages to be properly written.
+                // Flow control by not sending more when we already have many outstanding.
+                if (UidConfirmed && _resendBuffer.NonAcked.Count <= _settings.SysResendLimit) _writer.Tell(sequencedSend);
```

Comment:
The comment above this line explains the rationale for this change: don't attempt to send another `SystemMessage` right now if we still have `number of outstanding acks` > `RemoteSettings.SysResendLimit`.